### ISSUE TITLE
Creating a container for an existing workspace corrupts the git file(s) if there is at least one commit.

### ISSUE THREAD

**Issue by myermian** [microsoft/vscode-remote-release#2591]
- VSCode Version: 1.43.1 (user setup)
- Local OS Version: Windows_NT x64
- Remote OS Version: Debian GNU/Linux 10 (buster)
- Remote Extension/Connection Type: Docker
- Git Version: 2.17.1.windows.2 (installed with VS 2019)

**Steps to Reproduce**:
1. Clone an existing repository from a remote origin (Azure DevOps, Github, etc.) or alternatively create a repository on your local machine, initialize it, and make at least one commit.
2. Make a modification to the repository. Notice that the source control identifies the change properly. This is further verified by running the command line argument `git diff`.
4. Run `Remote-Containers: Add Development Container Configuration Files...`.
5. Run `Remote-Containers: Reopen in Container`.

It seems that when a folder already contains a git file with at least one commit, then building a container for the first time corrupts the git file. Within the container, the `git diff` command produces an error:

```
root@4fec1c5391fe:/workspaces/example# git diff
fatal: unable to read 3e759b75bf455ac809d0987d369aab89137b5689
```

**Work around**:
1. Create a folder and open that folder in VS Code.
2. Initialize the repository.
4. Run `Remote-Containers: Add Development Container Configuration Files...`. Notice that the source control identifies the change properly. This was tested with `C# (.NET Core 3.1)` and `Node.js 12 & Typescript`, so I don't think the container configuration definition which was selected matters?
5. Run `Remote-Containers: Reopen in Container`. Notice that the source control still identifies the change properly.
6. Add a remote origin with bash using the command `git remote add origin {url}`.
7. Pull the remote origin with bash using the command `git pull origin master`.

```
root@9d2053af16b7:/workspaces/example# git remote add origin https://github.com/microsoft/vscode-remote-release.git
root@9d2053af16b7:/workspaces/example# git remote -v
origin  https://github.com/microsoft/vscode-remote-release.git (fetch)
origin  https://github.com/microsoft/vscode-remote-release.git (push)
root@9d2053af16b7:/workspaces/example# git pull origin master
remote: Enumerating objects: 106, done.
remote: Total 106 (delta 0), reused 0 (delta 0), pack-reused 106
Receiving objects: 100% (106/106), 36.29 MiB | 7.16 MiB/s, done.
Resolving deltas: 100% (38/38), done.
From https://github.com/microsoft/vscode-remote-release
 * branch            master     -> FETCH_HEAD
 * [new branch]      master     -> origin/master
root@9d2053af16b7:/workspaces/example# 
```

Does this issue occur when you try this locally?: No
Does this issue occur when you try this locally and all extensions are disabled?: No

**Comment by Chuxel** [comment]
This likely is the same issue as #2571 - there is an upstream docker bug with 2.2.0.4 that prevents read-only files from working. 

You can try installing this patched version of Docker while we wait for a release: 

https://github.com/docker/for-win/issues/6016#issuecomment-600285551

**Comment by Chuxel** [comment]
This should be resolved in the recently released 2.2.0.5 version of Docker.
